File: DALSERVER/SchoolServer.cs
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using ALOS.Expand;
using ALOS.Model;

namespace ALOS.DALSERVER
{
    /// <summary>
    /// v00003#学校学院管理表
    /// </summary>
    [APIServer(" v00003#学校学院管理表")]
    public class SchoolServer : Help<SchoolType>
    {
        /// <summary>
        /// 获得所有的学院学校信息
        /// </summary>
        /// <returns></returns>
        [APIServer("获得所有的学院学校信息")]
        public List<SchoolType> SelectList()
        {
            return _help.Select<SchoolType>("select * from v00003");
        }
        /// <summary>
        /// 保存学院学校信息
        /// </summary>
        /// <param name="type"></param>
        /// <returns></returns>
        [APIServer("保存学院学校信息", "学校对象")]
        public SchoolType SaveSchool(SchoolType type)
        {
            return base.Save(type);
        }
        /// <summary>
        /// 根据学校名称和学院名称查找学校对象
        /// </summary>
        /// <param name="schoolName"></param>
        /// <param name="collegeName"></param>
        /// <returns></returns>
        public SchoolType ReaderSchoolId(string schoolName, string collegeName)
        {
            string sql =
                " select * from V00003 where ObjId in(select id from V00003 where ObjId=0 and Name=@schoolName) and Name=@CollegeName";
            return _help.Reader<SchoolType>(sql, new SqlParameter("CollegeName", collegeName),
                new SqlParameter("schoolName", schoolName));
        }

        /// <summary>
        /// 删除学校信息
        /// </summary>
        /// <param name="id"></param>
        [APIServer("删除学校信息", "学校ID")]
        public void DeleteSchool(int id)
        {
            base.Delete(id);
        }
        /// <summary>
        /// 根据OBJiD获得数据
        /// </summary>
        /// <param name="objid"></param>
        /// <returns></returns>
        [APIServer("根据OBJiD获得数据","objid")]
        public List<SchoolType> SelectTypeSchool(int objid)
        {
            string sql = "select * from v00003 where objid=@objid";
            return _help.Select<SchoolType>(sql, new SqlParameter("objid", objid));
        }
        /// <summary>
        /// 根据ID获得对象
        /// </summary>
        /// <param name="id"></param>
        /// <returns></returns>
        [APIServer("根据ID获得对象","Id")]
        public SchoolType RenderSchool(int id)
        {
            string sql = "select * from v00003 where id=@id";
            return _help.Reader<SchoolType>(sql, new SqlParameter("id", id));
        }
    }
}

File: Valuating/Resource/Scripts/dbserver/SchoolServer.js
// v00003#学校学院管理表
DB.SchoolServer = function() {
};
//获得所有的学院学校信息 		
DB.SchoolServer.SelectList=function(success){
var array=new Array();

DB.CreateWebLog("SchoolServer", "SelectList", array,success);
};
//保存学院学校信息 学校对象		参数(1) ALOS.Model.SchoolType
DB.SchoolServer.SaveSchool=function(type,success){
var array=new Array();
array[0]=type;

DB.CreateWebLog("SchoolServer", "SaveSchool", array,success);
};
//删除学校信息 学校ID		参数(1) System.Int32
DB.SchoolServer.DeleteSchool=function(id,success){
var array=new Array();
array[0]=id;

DB.CreateWebLog("SchoolServer", "DeleteSchool", array,success);
};
//根据OBJiD获得数据 objid		参数(1) System.Int32
DB.SchoolServer.SelectTypeSchool=function(objid,success){
var array=new Array();
array[0]=objid;

DB.CreateWebLog("SchoolServer", "SelectTypeSchool", array,success);
};
//根据ID获得对象 Id		参数(1) System.Int32
DB.SchoolServer.RenderSchool=function(id,success){
var array=new Array();
array[0]=id;

DB.CreateWebLog("SchoolServer", "RenderSchool", array,success);
};
